File: DSAlgorithms/Leetcode/stacks/assortment/c++/682-baseball_game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        std::stack<int> a_stack;
        for(int i = 0; i < operations.size(); i++)
        {
            std::string c = operations[i];
            if(c == "+")
            {
                std::cout<<a_stack.top();
                int right = a_stack.top();
                a_stack.pop();
                int left = a_stack.top();
                //std::cout<<left<<" : "<<right;
                a_stack.pop();
                a_stack.push(left);
                a_stack.push(right);
                a_stack.push(left + right);
            } 
            else if(c == "D")
            {
                int top = a_stack.top();
                a_stack.pop();
                a_stack.push(top);
                a_stack.push(2 * top);
            }
            else if(c == "C")
                a_stack.pop();
            else
            {
                a_stack.push(stoi(c));
                //std::cout<<a_stack.top()<<'\n';
            }
        }
        int result = 0;
        while(!a_stack.empty())
        {
            result += a_stack.top();
            a_stack.pop();
        }
        return result;
    }
};

File: dsa_algo/leetcode/ts/src/stacks/682-baseball_game.ts
const calPoints = (operations: string[]): number =>{
    const a_stack: number[] = [];
    for(let i = 0; i < operations.length; i++)
    {
        let c = operations[i];
        if(c === "+")
        {
            let right = a_stack[a_stack.length - 1];
            let left = a_stack[a_stack.length - 2];
            a_stack.push(left + right);
        }
        else if(c === 'D')
        {
            let top = a_stack[a_stack.length - 1];
            a_stack.push(top * 2);
        }
        else if(c === "C")
            a_stack.pop();
        else
            a_stack.push(Number(c));
    }
    let result = 0;
    while(a_stack.length > 0)
    {
        result += a_stack[a_stack.length - 1];
        a_stack.pop();
    }
    return result;
};